src/main/java/io/quarkus/search/app/quarkiverseio/QuarkiverseIOConfig.java

```diff
@@ -10,25 +10,32 @@
 @ConfigMapping(prefix = "quarkiverseio")
 public interface QuarkiverseIOConfig {
 
+    @WithDefault("github-artifact")
+    SourceType source();
+
+    enum SourceType {
+        NONE,
+        GITHUB_ARTIFACT,
+        ZIP
+    }
+
     Optional<Zip> zip();
 
     Optional<GithubArtifact> githubArtifact();
 
-    @WithDefault("true")
-    boolean enabled();
-
     @WithDefault("https://docs.quarkiverse.io/")
     URI baseUri();
 
     interface GithubArtifact {
         String token();
 
+        @WithDefault("quarkiverse/quarkiverse-docs")
         String repository();
 
-        //@WithDefault( "Publish website" )
+        @WithDefault("Publish website")
```

Comment:
ohh I remembered why I kept this commented out ... 

> SRCFG00014: The config property quarkiverseio.github-artifact.token is required but it could not be found in any config source

I think it's a bug in the config: if there's a default value even though we have `Optional<GithubArtifact>` it "sets" the default values and it results in `githubArtifact` being "partially-complete" and "non-empty" 😖 😕

Reply:
Awesome... I'll use `Optional#orElse` creatively, then.

Reply:
I just removed the use of `Optional` on config groups, since that doesn't seem to work as we expect.